### PR TITLE
Refactor WhiteListPermissions to not take array position into account

### PR DESCRIPTION
**Pull request by jseppi**
This PR is a semi-rewrite of the `WhiteListPermissions` class. That class used to check that _ONLY_ the penultimate IP in X-Forwarded-Hosts was in the list of whitelisted IPs. I've modified it to instead check that _ANY_ IP in that header is in the list of whitelisted IPs.

I also added tests since there were none.

### CODE REVIEW

File: api/tests/test_permissions.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        w = WhiteListPermission()
+        req = mock.MagicMock(
+            META={
+                'HTTP_X_FORWARDED_FOR': '5.5.5.5, 2.2.3.3',
+                'REMOTE_ADDR': '5.6.7.8'
+            }
+        )
+        self.assertFalse(w.has_permission(req, None))
+
+    def test_it_prefers_forwarded_for_to_remote_addr(self):
```
Comment (jseppi) [inline api/tests/test_permissions.py]: Not sure if the behavior this tests for is absolutely necessary, but it was how the original implementation of `WhiteListPermission` worked so I kept the same behavior.

File: api/permissions.py [hunk 2]
```diff
@@ -8,20 +8,23 @@ class WhiteListPermission(permissions.BasePermission):
...
+        ip_addresses = []
+
         forwarded = request.META.get('HTTP_X_FORWARDED_FOR')
         if forwarded:
             try:
-                ip_addr = forwarded.split(',')[-2].strip()
+                ip_addresses = [f.strip() for f in forwarded.split(',')]
             except:
-                ip_addr = forwarded
+                ip_addresses = [forwarded]
```
Comment (jseppi) [inline api/permissions.py]: I don't quite understand what the original logic here (the `except` block) is for. Seems like it would happen if `forwarded` is not a string, so it should probably be specified as an `AttributeError`. But I can't think of a case where `forwarded` would NOT be a string, so maybe the entire `except` block should be removed. Thoughts, @toolness?

Comment (toolness) [inline api/permissions.py]: Hmm, in the original code, it was taking the penultimate forwarded IP via `[-2]`, which would result in an `IndexError` if there was only zero or one comma-separated terms there. So I am guessing that `except` should have been specified as an `IndexError`?

Since you're not taking a particular index, though, it seems like you can get rid of the `try`/`except` entirely.

File: api/tests/test_permissions.py (hunk 1 above)
Comment (toolness) [inline api/tests/test_permissions.py]: Noice, gud idea.

File: api/permissions.py (hunk 2 above)
Comment (jseppi) [inline api/permissions.py]: ah yeah, guess it could have been that also! Will remove 👍

**Comment by toolness** [comment]
Noice, feel free to merge at will!

**Comment by jseppi** [comment]
image
